eng/pipelines/merge-docs-main-to-live.yml

```diff
@@ -0,0 +1,86 @@
+parameters:
+- name: WhatIfPreference
+  type: boolean
+  default: false
+
+- name: DocsRepos
+  type: object
+  default:
+    - RepoName: azure-docs-sdk-java
+      RepoOwner: Azure
+    - RepoName: azure-docs-sdk-python
+      RepoOwner: MicrosoftDocs
+    - RepoName: azure-docs-sdk-node
+      RepoOwner: MicrosoftDocs
+
+- name: BaseBranch
+  type: string
+  default: live
+
+jobs:
+  - job: SyncUp
+    pool:
+      vmImage: windows-2019
+    variables:
+      DocRepoLocation: $(Pipeline.Workspace)/docs
+    steps:
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          Paths: 
+          - /eng/pipelines/branch-syncup.yml
+      - ${{ each repo in parameters.DocsRepos }}:
+        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+          parameters:
+            SkipDefaultCheckout: true
+            Repositories:
+            - Name: ${{ repo.RepoOwner }}/${{ repo.RepoName }}
+              WorkingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+
+        - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+          parameters:
+            WorkingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+            DefaultBranchVariableName: ${{repo.RepoName}}
+
+        - pwsh: | 
+            # Fetch all branches, e.g. main, live
+            Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" fetch origin"
+            git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" fetch origin | Write-Host
+
+            # Live branch tracks remote live branch.
+            Write-Host "git checkout -b ${{ parameters.BaseBranch }} -t origin/${{ parameters.BaseBranch }}"
+            git checkout -b ${{ parameters.BaseBranch }} -t origin/${{ parameters.BaseBranch }} | Write-Host
+
+            # Last commit for live branch. Use for file comparison later on
+            $lastCommit = (git log -1 | Out-String) -replace "(?ms).*commit (\w+).*", '$1' 
+            Write-Host "The last commit of live branch: $lastCommit"
+            Write-Host "##vso[task.setvariable variable=LastCommit]$lastCommit"
+
+            # Merge changes from main branch.
+            Write-Host "git pull origin $(${{repo.RepoName}})"
+            git pull origin $(${{repo.RepoName}})
+
+            # Rebase to get sort of the commits.
+            Write-Host "git rebase origin/${{ parameters.BaseBranch }}"
+            git rebase origin/${{ parameters.BaseBranch }}
+          displayName: Switch to live branch and merge main
+          workingDirectory: $(DocRepoLocation)/${{repo.RepoName}}
+
+        - pwsh: | 
+            Write-Host "git diff --name-only $(LastCommit) -- docfx.json"
+            $diff = git diff --name-only $(LastCommit) -- docfx.json
```

Comment:
Is this the only place LastCommit is referenced? If so, I suggest combining the steps to avoid setting an intermediate pipeline variable.